pkg/services/hcloud/network/network.go

```diff
@@ -154,6 +159,24 @@ func (s *Service) Delete(ctx context.Context) error {
 }
 
 func (s *Service) findNetwork(ctx context.Context) (*hcloud.Network, error) {
+	// if an ID was provided we want to use the existing Network.
+	id := s.scope.HetznerCluster.Spec.HCloudNetwork.ID
+	if id != nil {
+		network, err := s.scope.HCloudClient.GetNetwork(ctx, *id)
+		if err != nil {
+			hcloudutil.HandleRateLimitExceeded(s.scope.HetznerCluster, err, "GetNetwork")
+			return nil, fmt.Errorf("failed to get network %d: %w", *id, err)
+		}
+
+		if network != nil {
+			if len(network.Subnets) > 1 {
+				return nil, fmt.Errorf("multiple subnets not allowed")
```

Comment:
this should not be an error, but we rather should set an appropriate condition on the HetznerCluster object, create an event, and return nil here.

Reply:
I hope I understood you correctly here. I set the NetworkReadyCondition to false in that case. Also what do you mean with "create an event"?

Reply:
@johannesfrey  didn't see your response here sorry. "record.Warnf()" would be appropriate here. that's actually something we should still add!